mbtiles/src/mbtiles.rs

```diff
@@ -39,6 +40,82 @@ pub struct Metadata {
     pub json: Option<JSONValue>,
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct LevelDetail {
+    pub zoom: u8,
+    pub count: u64,
+    pub smallest: u64,
+    pub largest: u64,
+    pub average: f64,
+    pub bbox: Bounds,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct Statistics {
+    pub file_path: String,
+    pub file_size: u64,
+    pub schema: MbtType,
+    pub page_size: u64,
+    pub level_details: Vec<LevelDetail>,
+    pub count: u64,
+    pub smallest: Option<u64>,
+    pub largest: Option<u64>,
+    pub average: f64,
+}
+
+impl Display for Statistics {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "File: {}", self.file_path).unwrap();
+        writeln!(
+            f,
+            "FileSize: {:.2}B",
+            SizeFormatterBinary::new(self.file_size)
+        )
+        .unwrap();
+        writeln!(f, "Schema: {}", self.schema).unwrap();
+        writeln!(
+            f,
+            "Page size: {:.2}B",
+            SizeFormatterBinary::new(self.page_size)
+        )
+        .unwrap();
+        writeln!(
+            f,
+            "|{:^9}|{:^9}|{:^9}|{:^9}|{:^9}|{:^9}|",
+            "zoom", "count", "smallest", "largest", "average", "bbox"
+        )
+        .unwrap();
+
+        for l in &self.level_details {
+            writeln!(
+                f,
+                "|{:^9}|{:^9}|{:^9}|{:^9}|{:^9}|{:^9}|",
+                l.zoom,
+                l.count,
+                format!("{:.2}B", SizeFormatterBinary::new(l.smallest)),
+                format!("{:.2}B", SizeFormatterBinary::new(l.largest)),
+                format!("{:.2}B", l.average as u64),
+                l.bbox
+            )
+            .unwrap();
+        }
+        if self.count != 0 {
+            writeln!(
+                f,
+                "|{:^9}|{:^9}|{:^9}B|{:^9}B|{:^9}B|",
```

Comment:
Centering numbers is a bad idea - hard to compare them. Best to right-align.  Also, don't unwrap - will crash on empty (you probably want to have a test with an empty database - easy to create -- see examples in `tests/mbtiles.rs`... something like this:

```rust
let mbt = Mbtiles::new("file:mbtiles_empty_stats?mode=memory&cache=shared").unwrap();
let cn = mbtiles.open().await.unwrap();
create_flat_tables(&mut cn).await.unwrap();
// now you have both an mbtiles struct and a connection to it that you can pass to the statistic function
```

Reply:
Thx! One of my motivation is your Reviewing. Really learn a lot.  : )

Reply:
Working on this.